### PR TITLE
MEN-6827: Support new CLI interface of the mender client

### PR DESCRIPTION
**Pull request by lluiscampos**
Changelog: In Mender client v4.0, the `mender` binary has been split into `mender-update` and `mender-auth`. Modify `mender-connect` to call the right binary to trigger `check-update` and `send-inventory` Ticket: MEN-6671

### CODE REVIEW

**Comment by lluiscampos** [comment]
Please review also the commit classification,

In my opinion it is a fix from the `mender-connect` point of view, which does not honor a minor release but that needs to be cherry-picked to all supported versions of `mender-connect` (so that they will work with latest client).

**Review by kacf — APPROVED** [review]
Agree with the classification.

**Comment by lluiscampos** [comment]
@mender-test-bot cherry-pick to:
* 2.1.x
* 2.0.x

File: session/menderclient.go [hunk 1]
```diff
@@ -35,6 +36,18 @@ func MenderClient() Constructor {
 	return func() SessionHandler { return f }
 }
 
+func getMenderClientBinaryPath() (string, error) {
+	candidates := []string{"/usr/bin/mender-update", "/usr/bin/mender"}
```
Comment (oleorhagen) [inline session/menderclient.go:40]: Q: Is it not better to check if they are in the `PATH` ? :thinking:

Comment (lluiscampos) [reply session/menderclient.go:40]: Maybe you are right. Something like [this](https://github.com/mendersoftware/mender-artifact/blob/master/utils/binpath.go#L29)?

Is it okey if I push this a bit down the road? I make note as a low priority.

Comment (oleorhagen) [reply session/menderclient.go:40]: Absolutely - it is not a regression in any way :)

Comment (oleorhagen) [reply session/menderclient.go:40]: > Maybe you are right. Something like [this](https://github.com/mendersoftware/mender-artifact/blob/master/utils/binpath.go#L29)?

Yes :)

**Pull request merged**